django_socio_grpc/servicer_proxy.py

```diff
@@ -1,130 +1,252 @@
-                        logger.info(
-                            f"Receive action {action} on service {self.service_class.__name__}"
-                        )
-                    # db connection state managed similarly to the wsgi handler
-                    db.reset_queries()
-                    # INFO - AM - 30/06/2021 - Need this in production environnement to avoid SSL end of files errors when too much connection on database
-                    close_old_connections()
-
-                    service_instance = self.create_service()
-                    service_instance.request = request
-                    service_instance.context = GRPCSocioProxyContext(context, action)
-                    service_instance.action = action
-                    service_instance.before_action()
-
-                    # INFO - AM - 05/05/2021 - getting the real function in the service and then calling it if necessary
-                    instance_action = getattr(service_instance, action)
-                    if asyncio.iscoroutinefunction(instance_action):
-                        instance_action = async_to_sync(instance_action)
-                    return instance_action(service_instance.request, service_instance.context)
-                except GRPCException as grpc_error:
-                    logger.error(grpc_error)
-                    context.abort(grpc_error.status_code, grpc_error.get_full_details())
-                except Exception as error:
-                    etype, value, tb = sys.exc_info()
-                    formatted_exception = traceback.format_exception(etype, value, tb)
-                    # No need to send it to µservices logging because we did it as exception with log_unhandled_exception
-                    logger.error("".join(formatted_exception), extra={"emit_to_server": False})
-                    grpcHandler = GRPCHandler()
-                    grpcHandler.log_unhandled_exception(etype, value, tb)
-                    context.abort(grpc.StatusCode.UNKNOWN, str(error))
-                finally:
-                    # INFO - AM - 30/06/2021 - Need this in production environnement to avoid SSL end of files errors when too much connection on database
-                    close_old_connections()
-                    pass
-
-            return handler
-
-    def create_service(self):
-        return self.service_class(**self.initkwargs)
+    def create_service(self, **kwargs):
+        return self.service_class(**self.initkwargs, **kwargs)
 
     def __getattr__(self, action):
         if not hasattr(self.service_class, action):
             raise Unimplemented()
 
-        return self.call_handler(action)
+        return self.get_handler(action)
 
-
-def close_old_connections():
-    for conn in db.connections.all():
-        if conn.connection is None:
-            continue
-        if conn.get_autocommit():
-            conn.close_if_unusable_or_obsolete()
+    def process_exception(self, exc, request: GRPCRequestContainer):
```

Comment:
add comment to explain that we need that function to process exception because of the middleware and that based on the same logic thn django one

Reply:
It's not because of the middlewares it's old code that has been moved to a method for clarity